app/controllers/sp_components_controller.rb

```diff
@@ -57,6 +58,27 @@ def destroy
     redirect_to admin_path(anchor: component&.component_type)
   end
 
+  def associate_service
+    component_present = SpComponent.exists?(params[:sp_component_id])
+    service = Service.find_by_id(params[:service_id])
+
+    if component_present && service.present?
+      @event = AssignSpComponentToServiceEvent.create(service: Service.find_by_id(params[:service_id]), sp_component_id: params[:sp_component_id])
+
+      if @event.valid?
+        redirect_to sp_components_path
```

Comment:
It could be better to redirect to the sp_component show page `sp_components_path(params[:sp_component_id])`